sound/soc/sof/pcm.c

```diff
@@ -366,27 +367,26 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 
 		/* fallthrough */
 	case SNDRV_PCM_TRIGGER_START:
-		if (spcm->stream[substream->stream].suspend_ignored) {
+		if (spcm->stream[substream->stream].d0i3_compatible &&
+		    sdev->D0i3_pipeline_count) {
```

Comment:
we can ignore this sdev->D0i3_pipeline_count check here, it has no chance to be 0 here, isn't it?

Reply:
Yes, it does. In the case when you first start the stream

Reply:
Oh, correct, that means we are in suspending.